server/nodejs/resources/js/index.js

```diff
@@ -0,0 +1,199 @@
+            survey_title: 'Sample Survey 6',
+            status: 'published',
+            program_id: 104,
+            period_start: '2024-10-15T08:00:00',
+            period_end: '2024-11-15T17:00:00',
+            respondents: { current: 30, total: 100 }
+        }
+    ];
+
+    const cardData = [
+        { title: 'Overall Respondents for <br> Surveys Deployed', data: 30/100 },
+        { title: 'Overall Missed <br> Surveys Deployed', data: 5 },
+        { title: 'Surveys Deployed', data: 3 }
+    ];
+
+
+    const searchInput = document.querySelector('#searchInput');
+    const unpublishedContainer = document.querySelector('#unpublishedSurveys');
+    const publishedContainer = document.querySelector('#publishedSurveys');
+    const cardDashboard = document.querySelector('.card-dashboard');
+
+    function renderSurveys(data) {
+        unpublishedContainer.innerHTML = '';
+        publishedContainer.innerHTML = '';
+
+        data.forEach(survey => {
+            const container = survey.status === 'unpublished' ? unpublishedContainer : publishedContainer;
+            const surveyItem = document.createElement('div');
+            surveyItem.className = 'survey-item';
+
+            // Create the HTML content for each survey item
+            surveyItem.innerHTML = `
+        <span class="survey-title">${survey.survey_title}</span>
+        <span class="survey-respondents">(${survey.respondents.current}/${survey.respondents.total})</span>
+        <div class="status ${survey.status === 'published' ? 'published' : ''}">
+            <span class="survey-status">${survey.status.charAt(0).toUpperCase() + survey.status.slice(1)}</span>
+        </div>
+        <button data-id="${survey.survey_id}" class="${survey.status === 'unpublished' ? 'edit-btn' : 'details-btn'}">
+            <img src="../../resources/images/${survey.status === 'unpublished' ? 'Edit' : 'Details'}.png" alt="${survey.status === 'unpublished' ? 'Edit' : 'Details'} Icon" />
+        </button>
+    `;
+
+            container.appendChild(surveyItem);
+        });
+    }
+
+    // TODO: Logic for the Three Cards (Below is just a test run)
+    // Function to create and render cards
+    function renderCards() {
+        cardDashboard.innerHTML = ''; // Clear any previous cards
+
+        cardData.forEach((card, index) => {
+            const cardElement = document.createElement('div');
+            cardElement.className = 'card';
+
+            // If it's card 1 (index 0), display dynamic current/total respondents for the most recent survey
+            let displayData = card.data;
+            if (index === 0) {
+                // Find the most recent published survey
+                const recentSurvey = surveyData
+                    .filter(survey => survey.status === 'published') // Only published surveys
+                    .sort((a, b) => new Date(b.period_end) - new Date(a.period_end))[0]; // Sort by most recent end date
+
+                if (recentSurvey) {
```

Comment:
Can you explain why this is an if statement when recentSurvey does not return a boolean value, and I don't see why this is done this way